src/mmu.c

```diff
@@ -25,20 +25,28 @@ static const uint8_t bootrom[256] = {
     0xF5, 0x06, 0x19, 0x78, 0x86, 0x23, 0x05, 0x20, 0xFB, 0x86, 0x20, 0xFE, 0x3E, 0x01, 0xE0, 0x50
 };
 
-ErrorCode mem_init(void) {
-    uint8_t *rom_bytes = get_rom_bytes();
-
+ErrorCode mmu_init(void) {
     mem = calloc(1, 0x10000);
+    uint8_t *rom_bytes = get_rom_bytes();
+    if (rom_bytes == NULL) {
+        LOG_ERROR("ROM-bytes buffer is NULL!\n");
```

Comment:
Why not exit here directly insteas of some lines further down?

Reply:
I think so too, using the exit macro here should do.

Reply:
I did this because we still want to initialize our memory even though no ROM is loaded, e.g. for unit testing
If we exit with an error code here, this would not represent the correct return value based on its semantics

Reply:
Yeah but in cases where we don't want the ROM to be initialized (e.g. unit tests) we can simply omit the call to `rom_init` as the function is only called in `main.c` in the productive code

Reply:
Exactly, then the if statement could be avoid completely I think 🤔 
It basically catches the case that `rom_load` wasn't called before `mmu_init`, but that's something we actually need for unit tests like you mentioned.

Reply:
Ah I'm seeing right now, that we actually copy some stuff from `rom_bytes` after the if statement. So we cannot just omit the if statement I'm afraid.

Reply:
I also just noticed that we could completely omit `mmu_init` for the unit tests, since the RAM is now stack-allocated. So the mmu_init actually takes care of ROM/BOOT stuff that unit tests do not require.

Reply:
Yeah, this might have resolved this issue altogether.
I'll try to figure it out when I merge the updated main into here